waitgroup_test.go

```diff
@@ -98,5 +98,46 @@ func TestWaitGroup(t *testing.T) {
 			require.Panics(t, wg.Wait)
 			require.Equal(t, int64(2), i.Load())
 		})
+
+		t.Run("is caught by waitsafe", func(t *testing.T) {
+			t.Parallel()
+			var wg WaitGroup
+			wg.Go(func() {
+				panic("super bad thing")
+			})
+			p := wg.WaitSafe()
+			require.Contains(t, p.Error(), "super bad thing", p.Error())
+		})
+
+		t.Run("one is caught by waitsafe", func(t *testing.T) {
+			t.Parallel()
+			var wg WaitGroup
+			wg.Go(func() {
+				panic("one bad thing")
+			})
+			wg.Go(func() {
+				panic("another bad thing")
+			})
+			p := wg.WaitSafe()
+			require.Contains(t, p.Error(), "bad thing", p.Error())
```

Comment:
This test seems slightly misleading. I think it might be more clear if the test more checked that the returned panic matches one of the panic values exactly. Something like:

```go
require.True(t, p.Value == "one bad thing" || p.Value == "another bad thing")
```

Reply:
Ah, good point. How about checking simply if p is non-nil?

I think at first I was thinking about adding a small sleep and checking that the first one to panic is the one that is returned, but that may possibly add flakiness, so I decided against it.

I'll update to a non-nil check for now, but let me know if you would like to be more explicit. Thanks!